File: 프로그래머스/2/42578. 의상/의상.py
from collections import defaultdict

def solution(clothes:list)->int:
    dic=defaultdict(int)
    
    for _,cate in clothes:
        dic[cate] +=1
    answer = 1
    
    for count in dic.values():
        # +1의 의미는 해당 종류를 선택안햇을때
        answer*=(count+1)
    # -1 은 아무것도 선택안할때는 빼야하니까
    return answer-1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.HashMap;
import java.util.Map;
   
class Solution {
        public int solution(String[][] clothes) {
            Map<String,Integer> combi=new HashMap<>();

            for (String[] clothe : clothes) {
                combi.put(clothe[1],combi.getOrDefault(clothe[1],0)+1);
            }
            
            int answer=1;
            
            for (Integer value : combi.values()) {
                answer*=(value+1);
            }
            return answer-1;
        }
    }